src/main/java/marquez/service/JobService.java

```diff
@@ -226,6 +216,39 @@ public Job createOrUpdate(
     }
   }
 
+  /**
+   * find the uuids for the datsets based on their namespace/name
+   *
+   * @param ids the namespace/name ids of the datasets
+   * @return their uuids
+   */
+  private List<UUID> findUuids(List<DatasetId> ids) {
+    // group per namespace since that's how we can query the db
+    Map<@NonNull NamespaceName, List<DatasetId>> byNamespace =
+        ids.stream().collect(groupingBy(DatasetId::getNamespace));
+    // query the db for all ds uuids for each namespace and combine them back in one list
+    return byNamespace.entrySet().stream()
+        .flatMap(
+            (e) -> {
+              String namespace = e.getKey().getValue();
+              List<String> names =
+                  e.getValue().stream()
+                      .map((id) -> id.getName().getValue())
+                      .collect(toImmutableList());
+              List<DatasetRow> results = datasetDao.findAllIn(namespace, names);
+              if (results.size() < names.size()) {
+                List<String> actual =
+                    results.stream().map(DatasetRow::getName).collect(toImmutableList());
+                throw new DatasetNotFoundException(
```

Comment:
We'll want to avoid throwing an exception in the service layer that is defined in the API layer. For now, maybe we just throw the catch-all exception `MarquezServiceException`?